.github/workflows/publish-to-test-pypi.yml

```diff
@@ -0,0 +1,74 @@
+name: Publish Python 🐍 distribution 📦 to TestPyPI
+
+on:
+  workflow_dispatch:  # Keep manual trigger
+    inputs:
+      rc_version:
+        description: 'RC version number (e.g., 1, 2, 3)'
+        required: true
+        type: string
+  schedule:
+    - cron: "0 0 * * *"  # Run every day at midnight
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - name: Get date
+      id: date
+      run: echo "date=$(date +'%Y%m%d')" >> $GITHUB_OUTPUT
+    - name: Update version for nightly
+      if: github.event_name == 'schedule'
+      run: |
+        # Assuming your version is in setup.py or pyproject.toml
+        sed -i 's/version="\([^"]*\)"/version="\1rc${{ steps.date.outputs.date }}"/' setup.py
```

Comment:
I think we could name it `nightly-${date}` -- should it have 0.0.63 prefix also? what's the typical use by other projects like pytorch for example?

Reply:
Changed to append .dev instead of rc for nightly following pytorch convension. E.g. `pip install "torch==2.2.0.dev20231015+cu121"`